### PR TITLE
Task2/jdbc: travel agency

### PR DESCRIPTION
**Pull request by qnislavnachev**
There is a travel agency that have database with 2 tables Trip and
People, the agency schedule trips and have few methods that return some
information about the trips and people

### CODE REVIEW

**Comment by qnislavnachev** [comment]
travel agency task need a review

File: src/agency/TourAgent.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    private void rollback() {
+        try {
+            connection.rollback();
+        } catch (SQLException e1) {
+            e1.printStackTrace();
+        }
+    }
+
+    private void execute(String query, Object o) throws SQLException {
+        if (o instanceof Person) {
```
Comment (tsony-tsonev) [inline src/agency/TourAgent.java]: this is not good, you can create two methods instead

File: src/agency/TravelAgency.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import core.City;
+import core.Person;
+
+import java.sql.Date;
+import java.sql.SQLException;
+import java.util.List;
+
+public interface TravelAgency {
+
+    void scheduleTrip(Person person, String city, Date from, Date to) throws SQLException;
```
Comment (tsony-tsonev) [inline src/agency/TravelAgency.java]: do not pass sql Date to the methods

File: src/core/Trip.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+    public Trip(int ID, Date arrivalDate, Date departureDate, String city) {
+        this.personID = ID;
+        this.arrivalDate = arrivalDate;
+        this.departureDate = departureDate;
+        this.city = city;
+    }
+
+    @Override
+    public String toString() {
```
Comment (tsony-tsonev) [inline src/core/Trip.java]: if you do not need the tostring methods remove them

File: src/agency/TourAgent.java (hunk 1 above)
Comment (qnislavnachev) [inline src/agency/TourAgent.java]: fixed

File: src/core/Trip.java (hunk 3 above)
Comment (qnislavnachev) [inline src/core/Trip.java]: removed  toString

File: jdbc.iml [hunk 4]
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module type="JAVA_MODULE" version="4">
```
Comment (qnislavnachev) [inline jdbc.iml]: removed file

File: src/agency/TravelAgency.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+import core.City;
+import core.Person;
+
+import java.sql.Date;
+import java.sql.SQLException;
+import java.util.List;
+
+public interface TravelAgency {
+
+    void scheduleTrip(Person person, String city, String from, String to) throws SQLException;
```
Comment (tsony-tsonev) [inline src/agency/TravelAgency.java]: I meant that you should use regular java Date instead of the sql Date, not that you should use strings

Comment (qnislavnachev) [inline src/agency/TravelAgency.java]: it is fixed now I am passing util date to the methods

**Comment by tsony-tsonev** [comment]
LGTM

**Pull request merged**